examples/fullstack/frontend/lib/services/api_types.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'api_types.g.dart';
+
+@JsonSerializable()
+class GreetingRequest {
+  final String? name;
+
+  GreetingRequest({this.name});
+
+  factory GreetingRequest.fromJson(Map<String, dynamic> json) =>
+      _$GreetingRequestFromJson(json);
+
+  Map<String, dynamic> toJson() => _$GreetingRequestToJson(this);
+
+  @override
+  bool operator ==(Object other) =>
+      other is GreetingRequest && other.name == name;
+
+  @override
+  int get hashCode => name.hashCode;
+}
+
+@JsonSerializable()
+class GreetingResponse {
+  final String salutation;
+  final String name;
+
+  GreetingResponse({required this.salutation, required this.name});
+
+  factory GreetingResponse.fromJson(Map<String, dynamic> json) =>
+      _$GreetingResponseFromJson(json);
+
+  Map<String, dynamic> toJson() => _$GreetingResponseToJson(this);
+
+  @override
```

Comment:
are these ever compared or put in a hashtable?

Reply:
These are artifacts from the basic JSON example where comparisons are made during testing. However, that's not the case here, so I can remove.